### PR TITLE
TreasureMap plugin

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR continues the series on interaction with ObservationPlans. In this case, it adds the ability to submit the plan's tiles to treasuremap.space.

### CODE REVIEW

File: skyportal/handlers/api/observation_plan.py [hunk 1]
```diff
@@ -641,3 +648,169 @@ async def get(self, observation_plan_request_id):
...
+            pointing["time"] = Time(obs.obstime, format='datetime').isot
+            pointing["depth"] = 0.0
+            pointing["depth_unit"] = "ab_mag"
+            pointings.append(pointing)
+
+        payload["pointings"] = pointings
+
+        url = urllib.parse.urljoin(TREASUREMAP_URL, 'api/v0/pointings')
+        r = requests.post(url=url, json=payload)
+        r.raise_for_status()
```
Comment (guynir42) [inline skyportal/handlers/api/observation_plan.py:742]: Should this be caught and converted to `self.error`?

File: skyportal/handlers/api/observation_plan.py [hunk 2]
```diff
@@ -641,3 +648,169 @@ async def get(self, observation_plan_request_id):
...
+        payload = {
+            "graceid": graceid,
+            "api_token": altdata['TREASUREMAP_API_TOKEN'],
+            "instrumentid": instrument.treasuremap_id,
+        }
+
+        baseurl = urllib.parse.urljoin(TREASUREMAP_URL, 'api/v0/cancel_all')
+        url = f"{baseurl}?{urllib.parse.urlencode(payload)}"
+        r = requests.post(url=url)
+        r.raise_for_status()
```
Comment (guynir42) [inline skyportal/handlers/api/observation_plan.py:814]: same here.

File: skyportal/services/test_server/test_api_server.py [hunk 3]
```diff
@@ -57,6 +57,33 @@ def get_cache_file():
...
+        for (submit_type, pattern) in patterns.items():
+            if re.search(pattern, uri) is not None:
+                return submit_type
+
+        return False
+
+    r1_is_treasuremap = is_treasuremap_request(r1_uri)
+    r2_is_treasuremap = is_treasuremap_request(r2_uri)
+
+    assert r1_is_treasuremap == r2_is_treasuremap and r1.method == r2.method
```
Comment (guynir42) [inline skyportal/services/test_server/test_api_server.py:84]: What happens if both `r1_is_treasuremap` and `r2_is_treasuremap` are `False`?

**Review by guynir42 — APPROVED** [review]
Minor comments / questions below.

File: skyportal/services/test_server/test_api_server.py (hunk 3 above)
Comment (mcoughlin) [reply skyportal/services/test_server/test_api_server.py:84]: Luckily the "treasuremapness" is also caught in  the individual get, post, etc. calls like:
        elif self.request.uri == "/api/v0/pointings":
            match_on = ["treasuremap"]
So double safety here I guess.

File: skyportal/handlers/api/observation_plan.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/handlers/api/observation_plan.py:817]: Added some custom error handling here.

File: skyportal/handlers/api/observation_plan.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/handlers/api/observation_plan.py:741]: Here too.

**Review by guynir42 — APPROVED** [review]
LGTM

**Pull request merged**